lib/connection.cpp

```diff
@@ -974,8 +976,23 @@ void Connection::Private::consumeToDeviceEvents(Events&& toDeviceEvents)
             outdatedUsers += event.senderId();
             encryptionUpdateRequired = true;
             pendingEncryptedEvents.push_back(std::make_unique<EncryptedEvent>(event.fullJson()));
-        }, [](const Event& e){
-            // Unhandled
+        }, [this](const KeyVerificationRequestEvent& event) {
+            auto session = new KeyVerificationSession(q->userId(), event, q, false, q);
+            emit q->newKeyVerificationSession(session);
+        }, [this](const KeyVerificationReadyEvent& event) {
+            emit q->incomingKeyVerificationReady(event);
+        }, [this](const KeyVerificationStartEvent& event) {
+            emit q->incomingKeyVerificationStart(event);
+        }, [this](const KeyVerificationAcceptEvent& event) {
+            emit q->incomingKeyVerificationAccept(event);
+        }, [this](const KeyVerificationKeyEvent& event) {
+            emit q->incomingKeyVerificationKey(event);
+        }, [this](const KeyVerificationMacEvent& event) {
+            emit q->incomingKeyVerificationMac(event);
+        }, [this](const KeyVerificationDoneEvent& event) {
+            emit q->incomingKeyVerificationDone(event);
+        }, [this](const KeyVerificationCancelEvent& event) {
+            emit q->incomingKeyVerificationCancel(event);
```

Comment:
I wonder if it makes sense to replace this switchboard with one signal to announce changes in the verification session state machine. By the way, you can only pass event (references to events, that is) through direct signal/slot connections - probably it's better to update the the verification session and pass a pointer to it in the signal instead?